src/Compilers/CSharp/Portable/Symbols/TypeSymbol.cs

```diff
@@ -153,11 +153,11 @@ protected override sealed Symbol OriginalSymbolDefinition
         /// (for example, interfaces), null is returned. Also the special class System.Object
         /// always has a BaseType of null.
         /// </summary>
-        internal abstract NamedTypeSymbol BaseTypeNoUseSiteDiagnostics { get; }
+        internal abstract NamedTypeSymbol GetBaseTypeNoUseSiteDiagnostics(bool ignoreNonNullTypesAttribute = false);
```

Comment:
>= false [](start = 107, length = 7)

A number of the implementations of this method leave off the optional part of the parameter, which was confusing to me as a reader.

Reply:
Most overloads are never called directly. Only the overloads that are directly called got a default value. Basically, this is the minimum set of default values to successfully compile without modifying callers.

Reply:
Sure, but my point is about the definitions. It was confusing because as I was reading the PR, the first definitions of GetBaseTypesNoUseSiteDiagnostics I happened upon were just defined as `GetBaseTypeNoUseSiteDiagnostics(bool ignoreNonNullTypesAttribute)`, which made no sense.

---
In reply to: [200710475](https://github.com/dotnet/roslyn/pull/28320#discussion_r200710475) [](ancestors = 200710475)